libvmaf/src/feature/cuda/integer_adm/adm_dwt2.cu

```diff
@@ -316,25 +317,26 @@ __device__ __forceinline__ void adm_dwt2_8_vert_hori_kernel(const T * d_picture,
     }
 
 
-#define DWT_8_VERT_HORI(v_shift, v_add_shift, v_rows_per_thread, h_shift, h_add_shift, tile_width, tile_height, type)                                                  \
-    __global__ void adm_dwt2_8_vert_hori_kernel_##v_shift##_##v_add_shift##_##v_rows_per_thread##_##h_shift##_##h_add_shift##_##tile_width##_##tile_height##_##type (  \
+#define DWT_8_VERT_HORI(v_rows_per_thread, h_shift, h_add_shift, tile_width, tile_height, type)                                                  \
+    __global__ void adm_dwt2_8_vert_hori_kernel_##v_rows_per_thread##_##h_shift##_##h_add_shift##_##tile_width##_##tile_height##_##type (  \
             const type * d_picture,                                                                                                                                    \
             cuda_adm_dwt_band_t dst,                                                                                                                                   \
             cuda_i4_adm_dwt_band_t i4_dwt2,                                                                                                                            \
             int w, int h,                                                                                                                                              \
             int src_stride, int dst_stride,                                                                                                                            \
+            int16_t v_shift, int32_t v_add_shift,                                                                                                                      \
             AdmFixedParametersCuda params) {                                                                                                                           \
-        adm_dwt2_8_vert_hori_kernel<v_shift, v_add_shift, v_rows_per_thread, h_shift, h_add_shift, tile_width, tile_height, type>(                                     \
+        adm_dwt2_8_vert_hori_kernel<v_rows_per_thread, h_shift, h_add_shift, tile_width, tile_height, type>(                                                           \
                 d_picture, dst, i4_dwt2,                                                                                                                               \
-                w, h, src_stride, dst_stride, params);                                                                                                                 \
+                w, h, src_stride, dst_stride, v_shift, v_add_shift, params);                                                                                           \
     }
```

Comment:
Is it necessary to move those values to an argument instead of a template argument ? It should not make a major impact though.

Reply:
v_shift should be set to the bit depth of the input and v_add_shift is 1 << (bit_depth - 1). We could have a function per common bit depth (10,12,16) but it seemed simpler to pass as an argument. There might be a trade off I'm missing though, would be very interested if there is!

Reply:
That makes sense. I'll run some experiments to see if there's any performance impact.

Reply:
I've run a benchmark of 240 frames of 3840x2160 10 bit content. On a 3060Ti I get a mean of 6.623s with the current pull request and 6.617s with the new arguments removed and hard coded.

I think the current pull request is a little simpler than the optimisation, so I'd suggest we keep the pull request as is. What do you think?